rust/cardano-blockchain-types/src/point.rs

```diff
@@ -0,0 +1,595 @@
+    /// ```
+    #[must_use]
+    pub fn hash_or_default(&self) -> Vec<u8> {
+        match &self.0 {
+            pallas::network::miniprotocols::Point::Specific(_, hash) => hash.clone(),
+            pallas::network::miniprotocols::Point::Origin => Vec::new(),
+        }
+    }
+
+    /// Checks if two `Point` instances are strictly equal.
+    /// Strict equality means both the slot number and hash must be identical.
+    ///
+    /// # Parameters
+    ///
+    /// * `b` - Another `Point` instance to compare against.
+    ///
+    /// # Returns
+    ///
+    /// A `bool` indicating whether the two `Point` instances are strictly equal.
+    ///
+    /// # Examples
+    ///
+    /// ```rs
+    /// use cardano_chain_follower::Point;
+    ///
+    /// let point1 = Point::new(42, vec![1, 2, 3]);
+    /// let point2 = Point::new(42, vec![1, 2, 3]);
+    /// assert!(point1.strict_eq(&point2));
+    ///
+    /// let point3 = Point::new(42, vec![1, 2, 3]);
+    /// let point4 = Point::new(43, vec![1, 2, 3]);
+    /// assert!(!point3.strict_eq(&point4));
+    /// ```
+    #[must_use]
+    pub fn strict_eq(&self, b: &Self) -> bool {
+        self.0 == b.0
+    }
+}
+
+impl Display for Point {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::result::Result<(), std::fmt::Error> {
+        if *self == Self::ORIGIN {
+            return write!(f, "Point @ Origin");
+        } else if *self == Self::TIP {
+            return write!(f, "Point @ Tip");
+        } else if *self == Self::UNKNOWN {
+            return write!(f, "Point @ Unknown");
+        }
+
+        let slot = self.slot_or_default();
+        let hash = self.hash_or_default();
+        if hash.is_empty() {
+            return write!(f, "Point @ Probe:{slot}");
+        }
+        write!(f, "Point @ {slot}:{}", hex::encode(hash))
+    }
+}
+
+impl From<pallas::network::miniprotocols::Point> for Point {
+    fn from(point: pallas::network::miniprotocols::Point) -> Self {
+        Self(point)
+    }
+}
+
+impl From<Point> for pallas::network::miniprotocols::Point {
+    fn from(point: Point) -> pallas::network::miniprotocols::Point {
+        point.0
+    }
+}
+
+impl PartialOrd for Point {
+    /// Implements a partial ordering based on the slot number
+    /// of two `Point` instances. It only checks the slot number for ordering.
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for Point {
+    /// Implements a total ordering based on the slot number
+    /// of two `Point` instances. It only checks the slot number for ordering.
+    fn cmp(&self, other: &Self) -> Ordering {
+        cmp_point(&self.0, &other.0)
+    }
+}
+
+impl PartialEq<u64> for Point {
+    /// Allows to compare a `SnapshotID` against `u64` (Just the Immutable File Number).
+    ///
+    /// Equality ONLY checks the Immutable File Number, not the path.
+    /// This is because the Filename is already the Immutable File Number.
```

Comment:
Unless I'm missing something, it is a wrong comment.

Reply:
Yes it needs correcting